### PR TITLE
Run fmt and clippy for rust code

### PR DESCRIPTION
**Pull request by youyuanwu**
## Description
Rust code should be formatted and linted to ensure consistency and quality.
Added fmt and clippy lint to cargo pipeline.

## Testing
NA

## Documentation
NA

### CODE REVIEW

File: .github/workflows/cargo.yml [hunk 1]
```diff
@@ -40,6 +40,13 @@ jobs:
     - name: Install Cargo
       if: runner.os == 'Linux'
       run: curl https://sh.rustup.rs -sSf | sh -s -- -y
+
```
Comment (nibanks) [inline .github/workflows/cargo.yml:43]: nit: please remove the extra spaces, since they're not used anywhere else.

File: src/lib.rs [hunk 2]
```diff
@@ -33,9 +33,9 @@ pub type BOOLEAN = ::std::os::raw::c_uchar;
 
 /// Family of an IP address.
 pub type AddressFamily = u16;
-pub const ADDRESS_FAMILY_UNSPEC: AddressFamily = c_types::AF_UNSPEC as u16;
-pub const ADDRESS_FAMILY_INET: AddressFamily = c_types::AF_INET as u16;
-pub const ADDRESS_FAMILY_INET6: AddressFamily = c_types::AF_INET6 as u16;
+pub const ADDRESS_FAMILY_UNSPEC: AddressFamily = c_types::AF_UNSPEC;
+pub const ADDRESS_FAMILY_INET: AddressFamily = c_types::AF_INET;
+pub const ADDRESS_FAMILY_INET6: AddressFamily = c_types::AF_INET6;
```
Comment (nibanks) [inline src/lib.rs:38]: Looks like this breaks on other platforms?

Comment (youyuanwu) [reply src/lib.rs]: Fixed by keeping the cast and suppressing the linting.

File: src/lib.rs [hunk 3]
```diff
@@ -143,8 +146,10 @@ mod status {
 #[cfg(target_os = "linux")]
 mod status {
     pub const QUIC_STATUS_SUCCESS: u32 = 0;
-    pub const QUIC_STATUS_PENDING: u32 = 0xFFFFFFFE; /// -2
-    pub const QUIC_STATUS_CONTINUE: u32 = 0xFFFFFFFF; /// -1
+    pub const QUIC_STATUS_PENDING: u32 = 0xFFFFFFFE;
+    /// -2
+    pub const QUIC_STATUS_CONTINUE: u32 = 0xFFFFFFFF;
+    /// -1
```
Comment (nibanks) [inline src/lib.rs:152]: Why isn't it allowing comments on the same line? I'm not a fan of this change.

**Review by nibanks — APPROVED** [review]
Thanks for making this change!

**Pull request merged**